lib/std/os.zig

```diff
@@ -5684,6 +5684,11 @@ pub const SendMsgError = SendError || error{
     /// The socket is not connected (connection-oriented sockets only).
     SocketNotConnected,
     AddressNotAvailable,
+
+    /// Returned when the arguments to the function are invalid
+    /// This typically signifies a bug in the code, but on linux it is also returned when sendTo is used with an address
+    /// that is not reachable by the bound address.
+    InvalidArgument,
```

Comment:
I believe the convention here would be to add an error like `UnreachableAddress` if the situation is as you describe.  There are a few other cases like this where some error codes are re-used, and if the only other cases are a result of code bugs, the best we can do is assume the code is written correctly and assume it's the non-buggy error.